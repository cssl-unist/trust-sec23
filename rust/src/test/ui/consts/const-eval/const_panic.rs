#![feature(const_panic)]
#![crate_type = "lib"]

const MSG: &str = "hello";

const Z: () = std::panic!("cheese");
//~^ ERROR any use of this value will cause an error

const Z2: () = std::panic!();
//~^ ERROR any use of this value will cause an error

const Y: () = std::unreachable!();
//~^ ERROR any use of this value will cause an error

const X: () = std::unimplemented!();
//~^ ERROR any use of this value will cause an error
//
const W: () = std::panic!(MSG);
//~^ ERROR any use of this value will cause an error

const Z_CORE: () = core::panic!("cheese");
//~^ ERROR any use of this value will cause an error

const Z2_CORE: () = core::panic!();
//~^ ERROR any use of this value will cause an error

const Y_CORE: () = core::unreachable!();
//~^ ERROR any use of this value will cause an error

const X_CORE: () = core::unimplemented!();
//~^ ERROR any use of this value will cause an error

const W_CORE: () = core::panic!(MSG);
//~^ ERROR any use of this value will cause an error
