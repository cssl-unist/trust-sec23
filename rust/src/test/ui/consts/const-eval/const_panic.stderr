error: any use of this value will cause an error
  --> $DIR/const_panic.rs:6:15
   |
LL | const Z: () = std::panic!("cheese");
   | --------------^^^^^^^^^^^^^^^^^^^^^-
   |               |
   |               the evaluated program panicked at 'cheese', $DIR/const_panic.rs:6:15
   |
   = note: `#[deny(const_err)]` on by default
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:9:16
   |
LL | const Z2: () = std::panic!();
   | ---------------^^^^^^^^^^^^^-
   |                |
   |                the evaluated program panicked at 'explicit panic', $DIR/const_panic.rs:9:16
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:12:15
   |
LL | const Y: () = std::unreachable!();
   | --------------^^^^^^^^^^^^^^^^^^^-
   |               |
   |               the evaluated program panicked at 'internal error: entered unreachable code', $DIR/const_panic.rs:12:15
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:15:15
   |
LL | const X: () = std::unimplemented!();
   | --------------^^^^^^^^^^^^^^^^^^^^^-
   |               |
   |               the evaluated program panicked at 'not implemented', $DIR/const_panic.rs:15:15
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:18:15
   |
LL | const W: () = std::panic!(MSG);
   | --------------^^^^^^^^^^^^^^^^-
   |               |
   |               the evaluated program panicked at 'hello', $DIR/const_panic.rs:18:15
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:21:20
   |
LL | const Z_CORE: () = core::panic!("cheese");
   | -------------------^^^^^^^^^^^^^^^^^^^^^^-
   |                    |
   |                    the evaluated program panicked at 'cheese', $DIR/const_panic.rs:21:20
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:24:21
   |
LL | const Z2_CORE: () = core::panic!();
   | --------------------^^^^^^^^^^^^^^-
   |                     |
   |                     the evaluated program panicked at 'explicit panic', $DIR/const_panic.rs:24:21
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:27:20
   |
LL | const Y_CORE: () = core::unreachable!();
   | -------------------^^^^^^^^^^^^^^^^^^^^-
   |                    |
   |                    the evaluated program panicked at 'internal error: entered unreachable code', $DIR/const_panic.rs:27:20
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:30:20
   |
LL | const X_CORE: () = core::unimplemented!();
   | -------------------^^^^^^^^^^^^^^^^^^^^^^-
   |                    |
   |                    the evaluated program panicked at 'not implemented', $DIR/const_panic.rs:30:20
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: any use of this value will cause an error
  --> $DIR/const_panic.rs:33:20
   |
LL | const W_CORE: () = core::panic!(MSG);
   | -------------------^^^^^^^^^^^^^^^^^-
   |                    |
   |                    the evaluated program panicked at 'hello', $DIR/const_panic.rs:33:20
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 10 previous errors

